### PR TITLE
Improvements and fixes for CMake

### PR DESCRIPTION
**Pull request by cmcquinn**
Changes that I made to get wolfBoot building reliably with `PULL_LINKER_DEFINES`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -140,7 +123,34 @@ add_option(
...
+add_option(
+    "PULL_LINKER_DEFINES"
+    "Pull partition addresses from the linker script instead of defining fixed addresses in target.h (default: disabled)"
+    "no"
+    "yes;no")
+
+# If PULL_LINKER_DEFINES is set, partition addresses and size info will come from the linker script,
+# so no need to pass them on the command line. These variables do need to be set for building the
+# test apps.
+if(NOT DEFINED PULL_LINKER_DEFINES AND NOT DEFINED BUILD_TEST_APPS)
```
Comment (jpbland1) [inline CMakeLists.txt:137]: are there any other cases besides building the test apps where we need these to be checked?

File: cmake/wolfboot.cmake [hunk 2]
```diff
@@ -75,15 +75,15 @@ function(gen_wolfboot_signed_image TARGET)
 
     # generate signed image
     add_custom_command(
-        OUTPUT ${TARGET}_signed.bin
-        DEPENDS ${INPUT_IMAGE} ${WOLFBOOT_SIGNING_PRIVATE_KEY}
+        OUTPUT ${TARGET}_v${VERSION}_signed.bin
```
Comment (jpbland1) [inline cmake/wolfboot.cmake:78]: does changing the output binary name interfere with any automated scripts that may expect the old name without the version?

**Comment by jpbland1** [comment]
had some questions but the changes themselves work and cmake compiles correctly.

**Comment by dgarske** [comment]
@cmcquinn making sure you saw the questions from John Bland's peer review?

File: cmake/wolfboot.cmake (hunk 2 above)
Comment (danielinux) [reply cmake/wolfboot.cmake:78]: hi @jpbland1 , no automated scripts are currently using Cmake, so they won't be affected. Also, this change seems to use the correct name format now (as in image_v1_signed.bin) for the output file of the signing operation

Comment (cmcquinn) [reply cmake/wolfboot.cmake:78]: @danielinux hit the nail on the head with his comment. This changes the cmake build system to use the name format actually produced by the signing tools. In our internal company fork of wolfBoot, which the cmake changes are originally based off of, I modified the signing tools to not include the version number in the output file name.

**Review by dgarske — APPROVED** [review]
Changes are approved. Awaiting approval of contributor agreement to merge.

**Review by dgarske — APPROVED** [review]
Thanks Cameron. The contributor agreement has been approved and put on file.

**Pull request merged**